### PR TITLE
Simplify accept helpers in TCK

### PR DESCRIPTION
**Pull request by boggle**

### CODE REVIEW

File: community/cypher/compatibility-suite/src/test/scala/cypher/feature/parser/expectedResultsParserTest.scala [hunk 1]
```diff
@@ -173,7 +175,7 @@ class expectedResultsParserTest extends ParsingTestSupport {
     parse(value) should accept(path(link1, link2))
   }
 
-  private def parse(value: String): ValueMatcher = {
+  private def parse(value: String): cypher.feature.parser.matchers.Matcher[AnyRef] = {
```
Comment (Mats-SX) [inline community/cypher/compatibility-suite/src/test/scala/cypher/feature/parser/expectedResultsParserTest.scala:178]: Is this change necessary?

**Comment by Mats-SX** [comment]
Sweet!

File: community/cypher/compatibility-suite/src/test/scala/cypher/feature/parser/expectedResultsParserTest.scala (hunk 1 above)
Comment (boggle) [inline community/cypher/compatibility-suite/src/test/scala/cypher/feature/parser/expectedResultsParserTest.scala:178]: Give it a try (or perhaps I can look later), I needed it at some point. This whole PR was not only about accept it also interacts badly with java type erasure and handling of boxed/unboxed primitives in scala, so I had to try quite a few things to get it right.

Comment (Mats-SX) [inline community/cypher/compatibility-suite/src/test/scala/cypher/feature/parser/expectedResultsParserTest.scala:178]: I'll look at it later. I think it's good enough to be merged for now.
